### PR TITLE
[Gardening] De-RST the Lexicon

### PR DESCRIPTION
**Pull request by CodaFi**

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci smoke test

File: docs/Lexicon.md [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+Serialized LLVM [IR](#IR).
+
+## build czar
+
+Apple term for "the person assigned to watch CI this week".
+
+## canonical SIL
+
+SIL after the
+[mandatory passes](#mandatory-passes-/-mandatory-optimizations) have run.
```
Comment (benrimmington) [inline docs/Lexicon.md:71]:
```suggestion
[mandatory passes](#mandatory-passes--mandatory-optimizations) have run.
```

File: docs/Lexicon.md [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+protocol. Represented in the compiler by the ProtocolConformance type at
+the AST level. See also [witness table](#witness-table).
+
+## contextual type
+
+1. The expected type for a Swift sub-expression based on the rest of the
+   statement. For example, in the statement `print(6 # 9)`, the contextual
+   type of the expression `6 # 9` is `Any`.
+2. The type of a value or declaration from inside a potentially generic
+   context. This type may contain [archetypes](#archetype)` and cannot be
```
Comment (benrimmington) [inline docs/Lexicon.md:111]:
```suggestion
   context. This type may contain [archetypes](#archetype) and cannot be
```

File: docs/Lexicon.md [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+
+1. "Problem Report": An issue reported in [LLVM's bug tracker](https://llvm.org/bugs/).
+See also [SR](#SR).
+2. "pull request"
+
+## primary file
+
+The file currently being compiled, as opposed to the other files that are
+only needed for context. See also
+[Whole-Module Optimization](#WMO-(whole-module-optimization)).
```
Comment (benrimmington) [inline docs/Lexicon.md:390]:
```suggestion
[Whole-Module Optimization](#wmo-whole-module-optimization).
```

File: docs/Lexicon.md [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+## Radar
+
+[Apple's bug-tracking system](https://bugreport.apple.com), or an issue reported 
+on that system.
+
+## raw SIL
+
+SIL just after being generated, not yet in a form that can be used for
+IR generation.
+See [mandatory passes](#mandatory-passes-/-mandatory-optimizations).
```
Comment (benrimmington) [inline docs/Lexicon.md:430]:
```suggestion
See [mandatory passes](#mandatory-passes--mandatory-optimizations).
```

File: docs/Lexicon.md [hunk 5]
```diff
@@ -0,0 +1,562 @@
...
+for review or merging.
+
+## witness
+
+The value or type that satisfies a protocol requirement.
+
+## witness table
+
+The SIL (and runtime) representation of a [conformance](#conformance); essentially a
+[vtable](#vtable-(virtual-dispatch-table)) but for a protocol instead of
```
Comment (benrimmington) [inline docs/Lexicon.md:552]:
```suggestion
[vtable](#vtable-virtual-dispatch-table) but for a protocol instead of
```

File: docs/Lexicon.md [hunk 6]
```diff
@@ -0,0 +1,562 @@
...
+a class.
+
+Only conceptually related to a [value witness table](#VWT-(value-witness-table)).
```
Comment (benrimmington) [inline docs/Lexicon.md:555]:
```suggestion
Only conceptually related to a [value witness table](#vwt-value-witness-table).
```

File: docs/Lexicon.md [hunk 7]
```diff
@@ -0,0 +1,562 @@
...
+
+A construct detailing how a particular type conforms to a particular
+protocol. Represented in the compiler by the ProtocolConformance type at
+the AST level. See also [witness table](#witness-table).
+
+## contextual type
+
+1. The expected type for a Swift sub-expression based on the rest of the
+   statement. For example, in the statement `print(6 # 9)`, the contextual
+   type of the expression `6 # 9` is `Any`.
```
Comment (benrimmington) [inline docs/Lexicon.md:109]:
```suggestion
   statement. For example, in the statement `print(6 * 9)`, the contextual
   type of the expression `6 * 9` is `Any`.
```

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci test Windows

File: docs/Lexicon.md [hunk 8]
```diff
@@ -0,0 +1,562 @@
...
+
+The feature that no uninitialized variables, constants, or properties will
+be read by a program, or the analysis pass that operates on SIL to
+guarantee this. This was 
+[discussed on Apple's Swift blog](https://developer.apple.com/swift/blog/?id=28).
+
+## DNM
+
+"Do not merge". Placed in PR titles where discussion or analysis is still
+ongoing.
```
Comment (benrimmington) [inline docs/Lexicon.md:150]: Outdated? [*Draft pull requests*](https://github.blog/changelog/2020-04-08-convert-pull-request-to-draft/)

File: docs/Lexicon.md [hunk 9]
```diff
@@ -0,0 +1,562 @@
...
+   code in a way that is easy for a compiler or tool to manipulate.
+2. "LLVM IR": a particular IR used by the LLVM libraries for optimization
+   and generation of machine code.
+
+## IUO (implicitly unwrapped optional)
+
+A type like Optional, but it implicitly converts to its wrapped type. If
+the value is `nil` during such a conversion, the program traps just as
+it would when a normal Optional is force-unwrapped. IUOs implicitly
+convert to and from normal Optionals with the same wrapped type.
```
Comment (benrimmington) [inline docs/Lexicon.md:221]: Outdated? [*Reimplementation of Implicitly Unwrapped Optionals*](https://swift.org/blog/iuo/)

**Comment by CodaFi** [comment]
Yeah, I'm not touching the content of these entries. That needs to be done in separate PRs.

**Comment by CodaFi** [comment]
After this has merged through, I have one more PR to get the simpler RST files converted to markdown. I'll tackle the big ones one at a time after that.

**Comment by benrimmington** [comment]
[Driver.md](https://github.com/apple/swift/blob/master/docs/Driver.md) links to Lexicon.rst

**Comment by CodaFi** [comment]
I'll catch it in a follow-up.

**Comment by CodaFi** [comment]
⛵

**Pull request merged**